cf/test/test_subsampling.py

```diff
@@ -0,0 +1,249 @@
+import atexit
+import datetime
+import faulthandler
+import os
+import tempfile
+import unittest
+
+faulthandler.enable()  # to debug seg faults and timeouts
+
+import cf
+
+n_tmpfiles = 1
+tmpfiles = [
+    tempfile.mkstemp("_test_subsampling.nc", dir=os.getcwd())[1]
+    for i in range(n_tmpfiles)
+]
+(tempfile,) = tmpfiles
+
+
+def _remove_tmpfiles():
+    """Remove temporary files created during tests."""
+    for f in tmpfiles:
+        try:
+            os.remove(f)
+        except OSError:
+            pass
+
+
+atexit.register(_remove_tmpfiles)
+
+
+class SubsampledTest(unittest.TestCase):
+    """Test management of constructs with underlying subsampled arrays."""
```

Comment:
The linting checks use a tool (`docformatter`, as listed on the output report) that checks the docstring summary line length, and this one is just too long (probably my fault from adding docstrings to the tests), so for a full linting pass if we can just tweak this to shorten it a bit. How about:

```suggestion
    """Test management of underlying subsampled arrays."""
```

?